sympy/solvers/ode/ode.py

```diff
@@ -706,6 +706,220 @@ def _helper_simplify(eq, hint, match, simplify=True, ics=None, **kwargs):
+
+    >>> eq = f(x).diff(x, 2) + cos(x)
+    >>> dsubs(eq, {x: t**2, f(x): g(t)})
+    cos(t**2) + (Derivative(g(t), (t, 2))/(2*t)
+    - Derivative(g(t), t)/(2*t**2))/(2*t)
+    >>> dsubs(eq, {x: t, f(x): g(t)**2})
+    2*g(t)*Derivative(g(t), (t, 2)) + cos(t)
+    + 2*Derivative(g(t), t)**2
+
+    To apply a transformation with number of new variables greater
+    than number of old variables, a list of all new variables must be
+    given. For example, a transformation `x = a t` would be done as -
+
+    >>> dsubs(eq, {x: a*t, f(x): g(t)}, [t])
+    cos(a*t) + Derivative(g(t), (t, 2))/a**2
+    >>> dsubs(eq, {x: t, f(x): h(a)*g(t)**2}, [t, g(t)])
+    2*g(t)*h(a)*Derivative(g(t), (t, 2))
+    + 2*h(a)*Derivative(g(t), t)**2 + cos(t)
+
+    Let us see how ``dsubs`` works for some typical cases using an example
+    of homogeneous ODEs which can be solved using the substitution
+    `v = \frac{y}{x}`.
+
+    >>> from sympy import symbols, Function, dsolve
+    >>> x, t = symbols('x t')
+    >>> y, v = symbols('y v', cls=Function)
+    >>> eq = y(x).diff(x) - (x**2 + y(x)**2)/(x*y(x))
+
+    Transform the homogeneous equation to a linear equation
+
+    >>> homeq = dsubs(eq, {x:t, y(x): t*v(t)}).simplify()
+    >>> homeq
+    t*Derivative(v(t), t) - 1/v(t)
+
+    Find the solutions for the transformed equation
+
+    >>> solt = dsolve(homeq)
+    >>> solt
+    [Eq(v(t), -sqrt(C1 + 2*log(t))), Eq(v(t), sqrt(C1 + 2*log(t)))]
+
+    Re-substitute `v = y x` to get the solutions to the
+    original equation
+
+    >>> sol = list(map(lambda sol: dsubs(sol, {v(t): y(x)/x, t: x}), solt))
+    >>> sol
+    [Eq(y(x)/x, -sqrt(C1 + 2*log(x))), Eq(y(x)/x, sqrt(C1 + 2*log(x)))]
+
+    Let us look at another example - The 2nd order Cauchy Euler equation.
+
+    >>> x, a, b, t = symbols('x a b t')
+    >>> y, phi = symbols('y phi', cls=Function)
+    >>> eq = x**2*y(x).diff(x, 2) + a*x*y(x).diff(x) + b*y(x)
+
+    Transform the equation using the substitutions `x = e^t`
+    and `y(x) = \phi(t)`
+
+    >>> from sympy import exp, log, logcombine
+    >>> eqtrans = dsubs(eq, {x: exp(t), y(x): phi(t)}).simplify()
+    >>> eqtrans
+    a*Derivative(phi(t), t) + b*phi(t) - Derivative(phi(t), t) + Derivative(phi(t), (t, 2))
+
+    The equation is now transformed to a constant-coefficient equation.
+
+    >>> solt = dsolve(eqtrans)
+    >>> solt
+    Eq(phi(t), C1*exp(t*(-a - sqrt(a**2 - 2*a - 4*b + 1) + 1)/2) + C2*exp(t*(-a + sqrt(a**2 - 2*a - 4*b + 1) + 1)/2))
+
+    Re-substitute for `\phi(t)` and `t` to get the solution for
+    the original equation.
+
+    >>> sol = logcombine(dsubs(solt, {phi(t): y(x), t: log(x)}), force=True)
+    >>> sol
+    Eq(y(x), C1*x**(-a/2 - sqrt(a**2 - 2*a - 4*b + 1)/2 + 1/2) + C2*x**(-a/2 + sqrt(a**2 - 2*a - 4*b + 1)/2 + 1/2))
+    """
+    from sympy.core.function import Subs, Function, Derivative, AppliedUndef
+    from sympy.core.symbol import Symbol
+    from sympy.integrals.integrals import Integral
```

Comment:
Why are all these imports delayed? Can they not go at the top of the module?

Reply:
These are actually redundant. I will remove them.